hack/jenkins/e2e.sh

```diff
@@ -37,28 +37,114 @@ else
+    # If updating the list of selected tests, be sure to update the
+    # configuration for the pull job as well.
+    : ${GINKGO_TEST_ARGS:=" --ginkgo.skip=Skipped|Addon|Etcd|Networking.*intra|Hostdir.*MOD|NetworkingNew|Density|Reboot|Restart|PD|Nodes\sResize|Nodes\sNetwork|Shell|ServiceAccounts|Elasticsearch|Services.*change\sthe\stype|Services.*identically\snamed|multiport\sendpoints|Services.*release.*load\sbalancer|Services.*functioning\sexternal\sload\sbalancer "}
+    : ${KUBE_GCE_INSTANCE_PREFIX:="e2e-test-parallel"}
+    : ${PROJECT:="kubernetes-jenkins"}
+    # Override GCE defaults.
+    NUM_MINIONS="6"
+    ;;
+
+  # Runs the flaky tests on GCE in parallel.
+  kubernetes-e2e-gce-parallel-flaky)
+    : ${E2E_CLUSTER_NAME:="parallel-flaky"}
+    : ${E2E_NETWORK:="e2e-parallel-flaky"}
+    : ${GINKGO_PARALLEL:="y"}
+    : ${GINKGO_TEST_ARGS:="--ginkgo.skip=Skipped|Addon|Reboot|Restart|Nodes\sResize ginkgo.focus=Networking.*intra|Hostdir.*MOD|PD|Shell|ServiceAccounts|Elasticsearch|Services.*change\sthe\stype|Services.*identically\snamed|multiport\sendpoints|Services.*release.*load\sbalancer|Services.*functioning\sexternal\sload\sbalancer"}
+    : ${KUBE_GCE_INSTANCE_PREFIX:="parallel-flaky"}
+    : ${PROJECT:="k8s-jkns-e2e-gce-prl-flaky"}
+    # Override GCE defaults.
+    NUM_MINIONS="4"
+    ;;
+
+  # Runs only the reboot tests on GCE.
+  kubernetes-e2e-gce-reboot)
+    : ${E2E_CLUSTER_NAME:="jenkins-gce-e2e-reboot"}
+    : ${E2E_DOWN:="false"}
+    : ${E2E_NETWORK:="e2e-reboot"}
+    : ${GINKGO_TEST_ARGS:=" --ginkgo.focus=Reboot "}
+    : ${KUBE_GCE_INSTANCE_PREFIX:="e2e-reboot"}
+    : ${PROJECT:="kubernetes-jenkins"}
+    ;;
+
+  # Runs the performance/scalability tests on GCE. A larger cluster is used.
+  kubernetes-e2e-gce-scalability)
+    : ${E2E_CLUSTER_NAME:="jenkins-gce-e2e-scalability"}
+    : ${E2E_NETWORK:="e2e-scalability"}
+    : ${GINKGO_TEST_ARGS:="--ginkgo.focus=Performance\ssuite|should\sbe\sable\sto\shandle "}
+    : ${KUBE_GCE_INSTANCE_PREFIX:="e2e-scalability"}
+    : ${PROJECT:="kubernetes-jenkins"}
+    # Override GCE defaults.
+    MASTER_SIZE="n1-standard-4"
+    MINION_SIZE="n1-standard-2"
+    MINION_DISK_SIZE="50GB"
+    NUM_MINIONS="100"
+    ;;
+
+  kubernetes-pull-build-test-e2e-gce)
+    : ${E2E_CLUSTER_NAME:="jenkins-pull-gce-e2e-${EXECUTOR_NUMBER}"}
+    : ${E2E_NETWORK:="pull-e2e-parallel-${EXECUTOR_NUMBER}"}
+    : ${GINKGO_PARALLEL:="y"}
+    # This list should match the list in kubernetes-e2e-gce-parallel. It
+    # currently also excludes a slow namespace test.
+    : ${GINKGO_TEST_ARGS:=" --ginkgo.skip=Skipped|Addon|Etcd|Networking.*intra|Hostdir.*MOD|NetworkingNew|Density|Reboot|Restart|PD|Nodes\sResize|Nodes\sNetwork|Shell|ServiceAccounts|Elasticsearch|Services.*change\sthe\stype|Services.*identically\snamed|multiport\sendpoints|Services.*release.*load\sbalancer|Services.*functioning\sexternal\sload\sbalancer|Namespaces\sDelete\s90\spercent\sof\s100\snamespace\sin\s150\sseconds "}
```

Comment:
ditto here, etc.—really want to have common lists of "flaky," "non-parallel," "provider-specific," etc. above that we can combine here. it'll be a bit annoying to make fully disjoint sets (question: is ginkgo.skip OK if you specify duplicates? I know it's OK if you specify things that don't exist. If you can specify dups then we won't need fully disjoint sets of tests and combing the lists will be easier)